classes/class-wc-connect-shipping-label.php

```diff
@@ -145,30 +145,33 @@ protected function get_all_packages() {
 				$formatted_packages[ $package_id ] = $package;
 			}
 
-			$service_id = 'usps'; //TODO: remove hardcoding
-			$predefined_packages_schema = $this->service_schemas_store->get_predefined_packages_schema_for_service( $service_id );
-			$enabled_predefined_packages = $this->settings_store->get_predefined_packages_for_service( $service_id );
-
-			foreach ( $predefined_packages_schema as $group ) {
-				foreach ( $group->definitions as $package ) {
-					if ( ! $package->is_flat_rate && ! in_array( $package->id, $enabled_predefined_packages ) ) {
-						continue;
+			$predefined_packages_schema = $this->service_schemas_store->get_predefined_packages_schema();
+			$enabled_predefined_packages = $this->settings_store->get_predefined_packages();
+
+			foreach ( $predefined_packages_schema as $service_id => $service_predefined_packages_schema ) {
+				$service_enabled_predefined_packages = isset( $enabled_predefined_packages[ $service_id ] ) ? $enabled_predefined_packages[ $service_id ] : array();
+				foreach ( $service_predefined_packages_schema as $group ) {
+					foreach ( $group->definitions as $package ) {
+						if ( ! $package->is_flat_rate && ! in_array( $package->id, $service_enabled_predefined_packages ) ) {
+							continue;
+						}
+
+						$formatted_packages[ $package->id ] = $package;
```

Comment:
I don't like to flatten this data structure this way (same as in the `get_package_lookup` method). This will break in subtle ways if packages from multiple carriers have the same ID. This should be more robust.